robottelo/datafactory.py

```diff
@@ -288,3 +289,51 @@ def valid_usernames_list():
         gen_string("alphanumeric"),
         gen_string("numeric"),
     ]
+
+
+def valid_http_credentials(url_encoded=None):
+    """Returns a list of valid credentials for HTTP authentication
+
+    :param url_encoded: flag for quoting special characters
+    :return: A list of dictionaries with user and password credentials
+    """
+    credentials = [
+        {u'login': 'admin', u'pass': 'changeme'},
+        {u'login': '@dmin', u'pass': 'changeme'},
+        {u'login': 'adm/n', u'pass': 'changeme'},
+        {u'login': 'admin2', u'pass': 'ch@ngeme'},
+        {u'login': 'admin3', u'pass': 'chan:eme'},
+        {u'login': 'admin4', u'pass': 'chan/eme'},
+        {u'login': 'admin5', u'pass': 'ch@n:eme'},
+        {u'login': '0', u'pass': 'mypassword'},
+        {u'login': '0123456789012345678901234567890123456789',
+         u'pass': 'changeme'},
+    ]
+    if url_encoded is True:
+        return [{u'login': quote(cred['login'], ''),
```

Comment:
what is the need for the second parameter '' for the quote function here?  Are you trying to quote character / too?